src/site/content/en/blog/native-hardware-fullscreen/index.md

```diff
@@ -0,0 +1,565 @@
+<button id="goFS">Go Fullscreen</button>
+<script>
+  var goFS = document.getElementById('goFS');
+  goFS.addEventListener(
+    'click',
+    function () {
+      var videoElement = document.getElementById('videoElement');
+      videoElement.requestFullscreen();
+    },
+    false,
+  );
+</script>
+```
+
+If your `<video>` element doesn't have the controls attribute defined,
+there's no way for the user to control the video once they are fullscreen. The
+recommended way to do this is to have a basic container that wraps the video and
+the controls that you want the user to see.
+
+```html
+<div id="container">
+  <video></video>
+  <div>
+    <button>Play</button>
+    <button>Stop</button>
+    <button id="goFS">Go fullscreen</button>
+  </div>
+</div>
+<script>
+  var goFS = document.getElementById('goFS');
+  goFS.addEventListener(
+    'click',
+    function () {
+      var container = document.getElementById('container');
+      container.requestFullscreen();
+    },
+    false,
+  );
+</script>
+```
+
+This gives you a lot more flexibility because you can combine the container
+object with the CSS pseudo selector (for example to hide the "goFS" button.)
+
+```html
+<style>
+  #goFS:-webkit-full-screen #goFS {
+    display: none;
+  }
+  #goFS:-moz-full-screen #goFS {
+    display: none;
+  }
+  #goFS:-ms-fullscreen #goFS {
+    display: none;
+  }
+  #goFS:fullscreen #goFS {
+    display: none;
+  }
+</style>
+```
+
+Using these patterns, you can detect when fullscreen is running and adapt your
+user interface appropriately, for example:
+
+- By providing a link back to the start page
+- By Providing a mechanism to close dialogs or travel backwards
+
+### Launching a page fullscreen from home screen
+
+Launching a fullscreen web page when the user navigates to it is not possible.
+Browser vendors are very aware that a fullscreen experience on every page load
+is a huge annoyance, therefore a user gesture is required to enter fullscreen.
+Vendors do allow users to "install" apps though, and the act of installing is a
+signal to the operating system that the user wants to launch as an app on the
+platform.
+
+Across the major mobile platforms it is pretty easy to implement using either
+meta tags, or manifest files as follows.
+
+#### iOS
+
+Since the launch of the iPhone, users have been able to install Web Apps to
+the home screen and have them launch as full-screen web apps.
+
+```html
+<meta name="apple-mobile-web-app-capable" content="yes" />
+```
+
+> If content is set to yes, the web application runs in full-screen mode;
+> otherwise, it does not. The default behavior is to use Safari to display web
+> content. You can determine whether a webpage is displayed in full-screen mode
+> using the window.navigator.standalone read-only Boolean JavaScript property.
+> <a href="https://developer.apple.com/library/safari/documentation/AppleApplications/Reference/SafariHTMLRef/Articles/MetaTags.html">Apple</a>
+
```

Comment:
```suggestion
<blockquote>
<p>If content is set to yes, the web application runs in full-screen mode; otherwise, it does not. The default behavior is to use Safari to display web content. You can determine whether a webpage is displayed in full-screen mode using the window.navigator.standalone read-only Boolean JavaScript property.</p>
<cite><a href="https://developer.apple.com/library/safari/documentation/AppleApplications/Reference/SafariHTMLRef/Articles/MetaTags.html">Apple</a></cite>
</blockquote>


```

Reply:
https://web.dev/handbook/web-dev-components/#block-quotes

Reply:
@rachelandrew added blockquote